### PR TITLE
VSphere: add dhclient.conf & require cluster in install config

### PR DESCRIPTION
**Pull request by patrickdillon**
This pull request ties up a couple of loose threads with VSphere IPI.

Adds the missing dhclient.conf file for bootstrap to use its own dns server. This was originally @jcpowermac's work. 

A bit more substantial is requiring the `cluster` field in the `install-config.yaml`. As part of the TF config we are reading in a cluster resource, so if the name of a cluster is not supplied, TF will crash. 

There is a related PR to update the VSphere UPI job to supply the newly required field: openshift/release#6836.

/cc jcpowermac
/cc mtnbikenc

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
We cannot make a previously not required field now required without bumping the API version.. as that would break API compatibility.

**Comment by patrickdillon** [comment]
> We cannot make a previously not required field now required without bumping the API version.. as that would break API compatibility.

I see. I will check whether it is possible to create a cluster if one is not specified.

File: pkg/types/vsphere/validation/platform.go [hunk 1]
```diff
@@ -41,5 +41,8 @@ func ValidatePlatform(p *vsphere.Platform, fldPath *field.Path) field.ErrorList
 		}
 	}
 
+	if len(p.Cluster) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("cluster"), "must specify the cluster"))
```
Comment (wking) [inline pkg/types/vsphere/validation/platform.go]: If this is required for IPI, but not for UPI, then we can leave it optional here (but maybe log a warning?).  And then die around [here][1] if the value is unset?  Also, I think @abhinavdahiya raised some good concerns with these new properties [back][2] [here][3].  Working out the consuming code when you add new properties would make it easier to get these sorts of wrinkles sorted earlier in development.

[1]: https://github.com/openshift/installer/blob/75ad924123661d417a3257bfbdc68beae5811959/pkg/asset/cluster/tfvars.go#L433
[2]: https://github.com/openshift/installer/pull/2875#discussion_r363070297
[3]: https://github.com/openshift/installer/pull/2875#issuecomment-570864578

**Comment by sdodson** [comment]
> We cannot make a previously not required field now required without bumping the API version.. as that would break API compatibility.

Because the current product documentation refers to this value as being required to match administrator configured resources in the vSphere cluster and the relatively low impact of breaking the API in this scenario we've consciously decided to move forward with requiring `Cluster`

**Comment by patrickdillon** [comment]
if we're moving forward with this, we have to prepare CI openshift/release#6836

**Comment by patrickdillon** [comment]
Also I think this does not address the VIPs which seem to also be required. If VIPs aren't included the failure is more subtle: terraform does not blow up like it does with cluster.

**Comment by sdodson** [comment]
/hold
sounds like there's more to sort out

**Comment by jcpowermac** [comment]
> Also this does not address the VIPs which seem to also be required. If VIPs aren't included the failure is more subtle: terraform does not blow up like it does with cluster.

@patrickdillon is this not enough?
https://github.com/openshift/installer/blob/master/pkg/types/vsphere/validation/platform.go#L32

**Comment by patrickdillon** [comment]
> @patrickdillon is this not enough?

No, I don’t think so, because this validation requires all or nothing. But we can’t allow nothing, right? I know that is the case on our internal dev vcenter.

**Comment by patrickdillon** [comment]
@abhinavdahiya can you help resolve this issue? 

To recap: if `cluster`, `APIVIP`, `DNSVIP`, and `IngressVIP` are not provided in a VSphere IPI install then installation will fail. It seems like they should be validated/required. As you pointed out, this should require an API bump. On the other hand, we think we're talking about breaking a very small number of users.

**Comment by abhinavdahiya** [comment]
> @abhinavdahiya can you help resolve this issue?
> 
> To recap: if `cluster`, `APIVIP`, `DNSVIP`, and `IngressVIP` are not provided in a VSphere IPI install then installation will fail. It seems like they should be validated/required. As you pointed out, this should require an API bump. On the other hand, we think we're talking about breaking a very small number of users.

The `*VIP` fields are required for self-hosted networking that is required for successful IPI installs but not necessary at all for UPI... so we need to ensure that we make these fields required only when cluster needs to be created by the installer, and technically there might a requirement for IPI that has external networking later on..

so one solution would be to require these values when create cluster is invoked..

**Comment by sdodson** [comment]
> so one solution would be to require these values when create cluster is invoked..

Is that acceptable? We could've done the same for `Cluster` but that seemed like an unacceptable solution at the time.

**Comment by patrickdillon** [comment]
> Is that acceptable? We could've done the same for Cluster but that seemed like an unacceptable solution at the time.

I remember that @jstuever brought up cluster-creation validation in discussion as something that has been discussed (or needed, can't remember exactly) for a long while, but I don't remember if we drew any conclusions on that.

I will create two separate cards. One for the validation and one for the dhclient.
Closing this but hope that doesn't discourage discussion.
/close

**Comment by patrickdillon** [comment]
@abhinavdahiya I created:
https://issues.redhat.com/browse/CORS-1354
https://issues.redhat.com/browse/CORS-1355

I hope these capture the requirements.

**Pull request closed without merging**